deeppavlov/models/ner/NER_model.py

```diff
@@ -0,0 +1,307 @@
+                                                     shape=[char_vocab_size, char_dim],
+                                                     trainable=True,
+                                                     initializer=xavier_initializer())
+                embedded_cnn_chars = tf.nn.embedding_lookup(tf_char_embeddings,
+                                                            char_ids_ph,
+                                                            name="embedded_cnn_chars")
+                conv1 = tf.layers.conv2d(inputs=embedded_cnn_chars,
+                                         filters=128,
+                                         kernel_size=(1, 3),
+                                         strides=(1, 1),
+                                         padding="same",
+                                         name="conv1",
+                                         kernel_initializer=xavier_initializer_conv2d())
+                conv2 = tf.layers.conv2d(inputs=conv1,
+                                         filters=128,
+                                         kernel_size=(1, 3),
+                                         strides=(1, 1),
+                                         padding="same",
+                                         name="conv2",
+                                         kernel_initializer=xavier_initializer_conv2d())
+                char_cnn = tf.reduce_max(conv2, axis=2)
+
+                self._input_features.append(char_cnn)
+
+        # ELMo
+        if elmo_dim is not None:
+            with tf.variable_scope("elmo_emb"):
+                padded_x_tokens_ph = tf.placeholder(tf.string, [None, None], name="padded_x_tokens")
+                self._xs_ph_list.append(padded_x_tokens_ph)
+
+                elmo = hub.Module("https://tfhub.dev/google/elmo/2", trainable=True)
+                emb = elmo(inputs={"tokens": padded_x_tokens_ph, "sequence_len": self.real_sent_lengths_ph},
+                           signature="tokens", as_dict=True)["elmo"]
+                elmo_emb = tf.layers.dense(emb, elmo_dim, activation=None)
+                self._input_features.append(elmo_emb)
+
+        features = tf.nn.dropout(tf.concat(self._input_features, axis=2), self._dropout_ph)
+
+        with tf.variable_scope("bi_lstm_words"):
+            cell_fw = tf.contrib.rnn.LSTMCell(lstm_hidden_size)
+            cell_bw = tf.contrib.rnn.LSTMCell(lstm_hidden_size)
+            (output_fw, output_bw), _ = tf.nn.bidirectional_dynamic_rnn(cell_fw, cell_bw, features,
+                                                                        sequence_length=self.real_sent_lengths_ph,
+                                                                        dtype=tf.float32)
+            self.output = tf.concat([output_fw, output_bw], axis=-1)
+
+            ntime_steps = tf.shape(self.output)[1]
+            self.output = tf.reshape(self.output, [-1, 2 * lstm_hidden_size])
```

Comment:
it might be not as fast as 
https://github.com/deepmipt/DeepPavlov/blob/master/deeppavlov/core/layers/tf_layers.py#L820

Reply:
I tried the `cudnn_bi_lstm` class and observed that there is no remarkable difference, even with the big datasets like OntoNotes.